xds/src/main/java/io/grpc/xds/XdsLoadBalancer.java

```diff
@@ -213,10 +276,13 @@ public void handleSubchannelState(Subchannel subchannel, ConnectivityStateInfo n
   @Override
   public void shutdown() {
     if (xdsLbState != null) {
-      XdsComms xdsComms = xdsLbState.shutdownAndReleaseXdsComms();
-      if (xdsComms != null) {
-        xdsComms.shutdownChannel();
+      if (lrsWorking) {
+        lrsClient.stopLoadReporting();
+        lrsClient = null;
+        lrsWorking = false;
       }
+      ManagedChannel channel = xdsLbState.shutdownAndReleaseChannel("Client shutdown");
```

Comment:
The message seems misleading.

Reply:
I don't even think we should have a message here. But this is because of [how `XdsComms` handles shutdown](https://github.com/grpc/grpc-java/blob/3432395119095e542a646aeafab6513fb7bf4a1a/xds/src/main/java/io/grpc/xds/XdsComms.java#L303). It takes a message and use it as the status description for `onError()`. This isn't quite necessary and we can do things like in [`Grpclb`](https://github.com/grpc/grpc-java/blob/3432395119095e542a646aeafab6513fb7bf4a1a/grpclb/src/main/java/io/grpc/grpclb/GrpclbState.java#L661). But, I was trying not to make excessive changes. If you want I can change it to something similar to `grpclb` (not sure if in this PR, although the change doesn't seem to be big).

Reply:
Why shouldn't have a message? A message provides more information in the log. It's different from the grpclb case you pointed to. Xds will not call `onComplete()`, in which case there is no message.

Reply:
Yeah, that's what I mean. Why would cancelling lb RPC invoking `onError()` instead of `onComplete`?

Reply:
You can have `shutdownAndReleaseChannel()` without argument, but in side `shutdownAndReleaseChannel()` create a message.

Reply:
`onComplete()` is just halfClose, can't guarantee the RPC to be closed.

Reply:
Then that really becomes making `XdsComms#shutdownLbRpc()` taking no argument and every call made through it will do `adsStream.cancelRpc("Client shutdown", null)`, which makes the message useless. I would prefer changing `XdsComms#AdsStream#cancelRpc()` to allow `message` as optional, just like `Grpclb`'s [`GrpclbState#LbStream#close()`](https://github.com/grpc/grpc-java/blob/3432395119095e542a646aeafab6513fb7bf4a1a/grpclb/src/main/java/io/grpc/grpclb/GrpclbState.java#L661).

Btw, why would you like to close lb RPC when receiving an invalid EDS response, instead of just skip processing that response?

Reply:
>I would prefer changing XdsComms#AdsStream#cancelRpc() to allow message as optional, just like Grpclb's GrpclbState#LbStream#close().

This was already discuss [here](https://github.com/grpc/grpc-java/pull/5294#discussion_r251974597). Again, a message is more informative than nothing in the log.


>Btw, why would you like to close lb RPC when receiving an invalid EDS response, instead of just skip processing that response?

Because in this case it's very likely to keep receiving invalid EDS responses. Close and go to fallback immediately is more appropriate.

Reply:
With offline discussion, we keep the `message` argument as we will close lb RPC with various reasons in `XdsLoadBalancer`.

Reply:
I would generally prefer a longer and more distinctive message, so that it can be easily located in the codebase if it ever becomes useful.

Reply:
Not for this one. I made message more informative on other places.